### ISSUE TITLE
All notifications are resent after each reconnection

### ISSUE THREAD

**Issue by dupuydauby** [MatthieuLemoine/electron-push-receiver#4]
Hello,

It seems that after a deconnection and reconnection:
_Connection to MCS server lost
MCS socket closed after 49 minutes
Trying to reconnect
MCS client connected_

I receive again all notifications since the beginning of the app.

In your blog you said you should have fixed it I think?
_But for now, I just skip them if they have already been received._

Tested on last version 1.2.0

**Comment by MatthieuLemoine** [comment]
That's weird because `electron-push-receiver` saves all the notification ids that has been received to avoid this issue. 

The ids are saved using [electron-config](https://github.com/sindresorhus/electron-store). 
You can see it using : 

```javascript
const Config = require('electron-config');
const persistentIds = config.get('persistentIds') 

console.log(persistentIds);
```

**Comment by MatthieuLemoine** [comment]
For now the list is not sent when reconnecting so the client will receive all/some notifications again but they are filtered client side. Already received notifications are [discarded by the client](https://github.com/MatthieuLemoine/push-receiver/blob/aaaac4ff96aca6bc2ebb4ef24e98cdc899234540/src/client/socket/index.js#L101-L103).

```javascript
// Already received
if (persistentIds.includes(object.persistentId)) {
  return;
}
```

**Comment by PedroKantar** [comment]
I reproduce this issue: after losing connection all the notifications are resent, looks like ```persistentIds``` is lost.
A candidate fix in push-receiver\src\client\socket\index.js would be to reload ```persistentIds" before [performing the retry](https://github.com/MatthieuLemoine/push-receiver/blob/aaaac4ff96aca6bc2ebb4ef24e98cdc899234540/src/client/socket/index.js#L35)

```
persistentIds = config.get('persistentIds') || [],
```
I'm going to test it...

**Comment by MatthieuLemoine** [comment]
I think I know why you have this issue.

In `retry`, we are using the persistentIds passed on the first connection attempt. So if new notifications are received during a connection, we will receive them again on subsequent retries.

How to reproduce :
 
- Connect
- Send a notification
- Disable internet connection
- Enable internet connection
- The notification is received again

Can you confirm this @PedroKantar and @dupuydauby ?

**Comment by MatthieuLemoine** [comment]
But you shouldn't receive all the notifications. Only those which have been received since the first connection attempt. All the previous notifications from previous sessions should not be received again.

**Comment by MatthieuLemoine** [comment]
The solution is simple, we just have to update the `persistentIds` array when we receive a new notification.

**Comment by PedroKantar** [comment]
"Can you confirm this @PedroKantar and @dupuydauby ?"
Yes I do. This exactly the sequence I observed, only the "delta" notifications between the last connect and the retry are received.

**Comment by PedroKantar** [comment]
@MatthieuLemoine : what do you think, as a first candidate fix, [at this line](https://github.com/MatthieuLemoine/push-receiver/blob/333626d663656d1a881142db0d9fb8b4ea066fcb/src/client/socket/index.js#L26) 
```
persistentIds = config.get('persistentIds') || [],
```
?

**Comment by MatthieuLemoine** [comment]
Well we don't have access to electron in push-receiver as it should work also in Node without electron. 

A better way will be to update the persistentIds array on each notification [here](https://github.com/MatthieuLemoine/push-receiver/blob/333626d663656d1a881142db0d9fb8b4ea066fcb/src/client/socket/index.js#L110)
```javascript
persistentIds.push(object.persistentId);

**Comment by PedroKantar** [comment]
OK, my previous fix worked, but I did not catch the "_should work also in Node without electron_" point, I'm going to test your candidate.. And get back asap.

**Comment by PedroKantar** [comment]
It works!

**Comment by MatthieuLemoine** [comment]
Great ! Can you submit a PR on push-receiver @PedroKantar ?

**Comment by PedroKantar** [comment]
Yes!

**Comment by MatthieuLemoine** [comment]
Fixed in [v1.2.2](https://github.com/MatthieuLemoine/electron-push-receiver/releases/tag/v1.2.2)

**Comment by dupuydauby** [comment]
Thanks guys, working very well!
